client.go

```diff
@@ -212,8 +232,16 @@ func (d *Dialer) Dial(urlStr string, requestHeader http.Header) (*Conn, *http.Re
 	// Get network dial function.
 	netDial := d.NetDial
 	if netDial == nil {
-		netDialer := &net.Dialer{Deadline: deadline}
-		netDial = netDialer.Dial
+		if d.NetDialContext == nil {
+			netDialer := &net.Dialer{Deadline: deadline}
+			netDial = func(network, addr string) (net.Conn, error) {
+				return netDialer.DialContext(ctx, network, addr)
+			}
+		} else {
+			netDial = func(network, addr string) (net.Conn, error) {
+				return d.NetDialContext(ctx, network, addr)
+			}
+		}
```

Comment:
Flatten the nested logic.

     var netDial func(network, addr string) (net.Conn, error)
     if d.NetDialContext != nil {
          netDial = func( ...
     } else if d.NetDial != nil {
           netDial = func( ...
      } else {
	    netDialer := &net.Dialer{Deadline: deadline}
	    netDial = netDialer.Dial
      }